libshaderc/include/shaderc.hpp

```diff
@@ -125,6 +126,35 @@ class CompileOptions {
     shaderc_compile_options_set_generate_debug_info(options_);
   }
 
+  // Define an includer interface, on which libshaderc will invoke the methods
+  // to resolve the full path and content of the file to be included, and clean
+  // up the resources used for the including process. Client code should
+  // implement this interface and configure the includer instance then set it to
+  // libshaderc through following API.
+  class Includer {
+   public:
+    virtual shaderc_includer_response* GetInclude(
+        const char* filename) = 0;
+    virtual void ReleaseInclude(shaderc_includer_response* data) = 0;
+  };
+
+  // Sets the includer instance for libshaderc to call on when resolving
```

Comment:
It is at fairly obvious from the interface, but might be worth explicitly mentioning that the ownership of the includer is passed to the CompileOptions, and will be destroyed with the CompileOptions.